### PR TITLE
[CM-380] Support HTML message type with rich messages

### PR DESCRIPTION
**Pull request by Sunilkumarr**
## Summary
* Added the support for showing the HTML message in case of rich messages
* Deleted  `ALKMyMessageView.swift ` and  `ALKFriendMessageView.swift` as some of the rich messages use a different class for showing message view and used the commons for all rich message text message `MessageView` from Rich message kit 

Screenshots of rich messages with HTML text : 

1. Suggested Quick reply and quick reply link button :

IMG_0017

2. image-rich  : 

IMG_0018

3. FAQ : 
IMG_0026

4. Card :
IMG_0024

5. List template ; 
IMG_0025

6. Form template : 
IMG_0023

## Motivation
Before when we send content Type 3  HTML and in metadata, there would be another rich message payload like a quick reply or any other in that case this metadata payload rich message were not displaying in UI 

## Testing
* Tested by sending content Type 3  HTML and with rich message payload and checked if it's displaying
* tested by sending without content-type 3 and with   rich message payload and checked if it's displaying

### CODE REVIEW

File: RichMessageKit/Models/Message.swift [hunk 1]
```diff
@@ -48,4 +65,7 @@ public struct Message {
         }
         return false
     }
+    /// Content type of the message.
+    public var contentType: Int16
```
Comment (mukeshthawani) [inline RichMessageKit/Models/Message.swift:69]: Type should be: `ContentType`

File: RichMessageKit/Utilities/MessageViewSizeCalculator.swift [hunk 2]
```diff
@@ -8,9 +8,24 @@
...
     }
+
+    func height(_ textView: UITextView,
+                        attributedText: NSAttributedString,
+                        maxWidth: CGFloat,
+                        padding: Padding) -> CGFloat {
+        let messageWidth = maxWidth - (padding.left + padding.right)
+         textView.attributedText = attributedText
+        let size = textView.sizeThatFits(CGSize(width: messageWidth, height: CGFloat.greatestFiniteMagnitude))
+         return ceil(size.height) + padding.top + padding.bottom
```
Comment (mukeshthawani) [inline RichMessageKit/Utilities/MessageViewSizeCalculator.swift:29]: There's an extra space in the front, make sure to format all the changes.

File: RichMessageKit/Views/core/MessageView.swift [hunk 3]
```diff
@@ -66,28 +109,45 @@ public class MessageView: UIView {
...
+            return
+        }
     }
 
-    /// It calculates height for `MessageView` based on the text passed and maximum width allowed for the view.
-    ///
+    /// It calculates height for `MessageView` based on the text passed and maximum width allowed for the view
     /// - Parameters:
-    ///   - text: Text set in messageView.
+    ///   - model: it will have message model  set in messageView.
```
Comment (mukeshthawani) [inline RichMessageKit/Views/core/MessageView.swift:144]: It should be: "Message for which height is to be calculated. "

File: RichMessageKit/Views/core/MessageView.swift [hunk 4]
```diff
@@ -98,10 +158,34 @@ public class MessageView: UIView {
...
         }
-        return MessageViewSizeCalculator().rowHeight(text: model,
-                                                     font: font,
-                                                     maxWidth: maxWidth,
-                                                     padding: padding)
+
+        switch model.contentType {
+        case Message.ContentType.text.rawValue:
+            self.dummyMessageView.font = font
+            return MessageViewSizeCalculator().height(dummyMessageView, text: model.text ?? "", maxWidth: maxWidth, padding: padding) + 3
```
Comment (mukeshthawani) [inline RichMessageKit/Views/core/MessageView.swift:165]: We can move the constant value(3) to `Padding`.

File: Sources/Models/ALKMessageModel.swift [hunk 5]
```diff
@@ -64,9 +64,11 @@ public protocol ALKMessageViewModel {
     var isReplyMessage: Bool { get }
     var metadata: [String: Any]? { get }
     var source: Int16 { get }
+    var contentType : Int16 { get }
```
Comment (mukeshthawani) [inline Sources/Models/ALKMessageModel.swift:67]: Type should be: `Message.ContentType`

**Review by mukeshthawani — COMMENTED** [review]
In case of card, list and form message type, text view and card view are not aligned properly. They should be left aligned, and their padding should be same.

File: RichMessageKit/Models/Message.swift (hunk 1 above)
Comment (Sunilkumarr) [reply RichMessageKit/Models/Message.swift]: Changed to ContentType

File: RichMessageKit/Utilities/MessageViewSizeCalculator.swift (hunk 2 above)
Comment (Sunilkumarr) [reply RichMessageKit/Utilities/MessageViewSizeCalculator.swift]: formated

File: RichMessageKit/Views/core/MessageView.swift (hunk 3 above)
Comment (Sunilkumarr) [reply RichMessageKit/Views/core/MessageView.swift]: Chnaged

File: RichMessageKit/Views/core/MessageView.swift (hunk 4 above)
Comment (Sunilkumarr) [reply RichMessageKit/Views/core/MessageView.swift]: Moved

File: Sources/Models/ALKMessageModel.swift (hunk 5 above)
Comment (Sunilkumarr) [reply Sources/Models/ALKMessageModel.swift]: changed

**Comment by Sunilkumarr** [comment]
Updated the screenshots

**Review by mukeshthawani — APPROVED** [review]

**Pull request merged**